src/vm/diagnosticserver.cpp

```diff
@@ -189,6 +195,12 @@ bool DiagnosticServer::Shutdown()
     }
     CONTRACTL_END;
 
+    // COMPlus_EnableDiagnostics==0 disables diagnostics so there is nothing to do here.
```

Comment:
This is not necessary. If it was not initialized, `s_pIpc` will be null and this whole thing will be no-op anyway.